tests/unit/__init__.py

```diff
@@ -60,3 +63,66 @@ def setUpClass(cls):
                 '__unittest_skip_why__',
                 f'Skipping this test case due to the "{skip_backend_tests_env_var}" being true',
             )
+
+
+class DatabaseBackendDependentTestCase(BackendDependentTestCase):
+    """
+    Extension of `BackendDependentTestCase` that sets a default value for `self.app.conf.database_url` based on the
+    `TEST_DATABASE_CELERY_RESULT_BACKEND_CONNECTION_STRING` environment variable. If no such environment variable
+    exists, the setup will assume a localhost connection to Postgres.
+    """
```

Comment:
I struggle with how to structure our unit tests here, particularly for test classes that require a live backend data store in order to work properly. The more backends we support, the more dependencies on external processes we will encounter. For example, I could see us needing Postgres, MongoDB, Redis, Memcached, RabbitMQ, and on and on in order to run a comprehensive unit test suite. I know that in other projects, we've used tools like embedded RabbitMQ or embedded Postgres so that our unit test suite can run with complete autonomy and no dependencies on docker containers running those dependencies alongside the test process. But installing those seemed like overkill here, given that upstream Flower currently does not have a solution to that issue. 

As long as we're ok with adding a bunch of additional containers to `docker-compose.yml`, we should be able to run these test successfully. However, I'm not convinced that the upstream Flower project will want to have all those dependencies. Just something to think about for future work.

Reply:
> As long as we're ok with adding a bunch of additional containers to docker-compose.yml, we should be able to run these test successfully. However, I'm not convinced that the upstream Flower project will want to have all those dependencies. Just something to think about for future work.

The more sustainable approach is to use docker-py, [here's an example or what I mean](https://github.com/shipwell/fastapi-microservice-template/blob/657899475b765d78e01d3452ea41453dfecafcbc/tests/test_generated_api.py#L29-L64).

If you go this route it might be better to look into [docker on whales](https://github.com/docker/docker-py/pull/2989#issuecomment-1179720247), the docker-py library is kind of a shit show, it languished for literal years. Like a full 12mo with a very high security vulnerability active before they fixed it.

Reply:
this is at the heart of why I prefer to say "automated tests" not "unit tests"
i want integration tests to be automated, which often means lots of nasty scripting,
scripting that falls outside of the realm of what unittest.TestCase is good at

a great place to start is by documenting manal steps taken to perform an integration test by hand,

for scripting against docker-compose or docker, docker:dind works totally great, i can pair up and give a demo of that any time, its a bit technical but it makes integrating against docker with predictable state possible.